src/components/routing/routers/AppRouter.js

```diff
@@ -43,19 +40,19 @@ const AppRouter = () => {
 
         {/* user guard */}
         <Route exact path={"/users/:id"}>
-          <UserGuard>
+          <GameGuard>
             <UserPage/>
-          </UserGuard>
+          </GameGuard>
```

Comment:
did you include the game guard here again in order to trigger the check for the token? if so, simple and clean fix! could have probably also used useEffect on history or push url events, but this works too